core/src/main/java/org/infinispan/globalstate/impl/EmbeddedLocalConfigurationManager.java

```
@@ -0,0 +1,167 @@
+      // Remove cache configuration and remove it from the computed cache name list
+      globalComponentRegistry.getComponent(ConfigurationManager.class).removeConfiguration(name);
+      // Remove cache from dependency graph
+      //noinspection unchecked
+      globalComponentRegistry.getComponent(DependencyGraph.class, CACHE_DEPENDENCY_GRAPH).remove(name);
+   }
+
+   public void loadAll() {
+      // Load any persisted configs
+      try (FileInputStream fis = new FileInputStream(getPersistentFile())) {
+         ConfigurationBuilderHolder holder = parserRegistry.parse(fis);
+         for (Map.Entry<String, ConfigurationBuilder> entry : holder.getNamedConfigurationBuilders().entrySet()) {
+            String name = entry.getKey();
+            Configuration configuration = entry.getValue().build();
+            // The cache was persisted, it still needs to be
+            globalConfigurationManager.createCache(name, configuration, EnumSet.of(CacheContainerAdmin.AdminFlag.PERSISTENT));
+         }
+      } catch (FileNotFoundException e) {
+         // Ignore
+      } catch (IOException e) {
+         throw new CacheConfigurationException(e);
+      }
+   }
+
+   private void storeAll() {
+      try {
+         File temp = File.createTempFile("caches", null, new File(globalConfiguration.globalState().persistentLocation()));
+         Map<String, Configuration> configurationMap = new HashMap<>();
+         for (String cacheName : persistentCaches) {
+            configurationMap.put(cacheName, cacheManager.getCacheConfiguration(cacheName));
+         }
+         try (FileOutputStream f = new FileOutputStream(temp)) {
+            parserRegistry.serialize(f, null, configurationMap);
+         }
+         FileLock lock = null;
+         try (FileOutputStream lockFile = new FileOutputStream(getPersistentFileLock())) {
+            lock = lockFile.getChannel().lock();
+            temp.renameTo(getPersistentFile());
```

IMO, it should "warn" if the rename (delete below) fails.

---

I'd prefer to throw an exception.

---

are you throw the exception in this PR or another? :)